core/tests/test_utils.py

```diff
@@ -592,8 +593,25 @@ def mock_search(
         terms = body['query']['bool']['must']
 
         for f in filters:
-            for k, v in f['match'].items():
-                result_docs = [doc for doc in result_docs if doc[k] in v]
+            # For processing 'doc[k] in v', doc[k] can only be of type string if
+            # v is a string.
+            if index == blog_services.SEARCH_INDEX_BLOG_POSTS:
+                for k, v in f['match'].items():
+                    # Tags field in blog post doc in blog post search index is
+                    # of type list(str) under which the blog post can be
+                    # classified. 'v' is a single tag which if present in the
+                    # tags field list, the doc should be returned. Therefore,
+                    # we check using 'v in doc[k]'.
```

Comment:
what is block post doc? Are you talking about `doc` variable? If yes then add quotes as it is confusing otherwise.

Reply:
Done